src/components/FolderPicker/FolderPickerTopbar.jsx

```diff
@@ -1,49 +1,106 @@
 import PropTypes from 'prop-types'
-import React from 'react'
+import React, { useCallback } from 'react'
 
-import { useQuery } from 'cozy-client'
+import { useClient } from 'cozy-client'
 import Icon from 'cozy-ui/transpiled/react/Icon'
 import IconButton from 'cozy-ui/transpiled/react/IconButton'
 import FolderAddIcon from 'cozy-ui/transpiled/react/Icons/FolderAdd'
-import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
+import Typography from 'cozy-ui/transpiled/react/Typography'
 import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
 
+import { getParentFolderFromPath } from './helpers'
 import BackButton from 'components/Button/BackButton'
+import { ROOT_DIR_ID } from 'constants/config'
 import Topbar from 'modules/layout/Topbar'
-import { getBreadcrumbPath } from 'modules/move/helpers'
-import Breadcrumb from 'modules/navigation/Breadcrumb/Breadcrumb'
+import { useNextcloudInfos } from 'modules/nextcloud/hooks/useNextcloudInfos'
 import { buildOnlyFolderQuery } from 'modules/queries'
 
 const FolderPickerTopbar = ({
   navigateTo,
-  folderId,
+  folder,
   showFolderCreation,
+  showNextcloudFolder,
   canCreateFolder
 }) => {
   const { t } = useI18n()
-  const { isMobile } = useBreakpoints()
-  const folderQuery = buildOnlyFolderQuery(folderId)
-  const { fetchStatus, data } = useQuery(
-    folderQuery.definition,
-    folderQuery.options
-  )
+  const client = useClient()
```

Comment:
The additions to this component seem quite heavy.

Helpers could be extracted
```js
const getParentFolder = async (folder, client) => {
  const parentFolderQuery = buildOnlyFolderQuery(folder.dir_id);
  const parentFolder = await client.fetchQueryAndGetFromState({
    definition: parentFolderQuery.definition(),
    options: parentFolderQuery.options
  });
  return parentFolder.data;
};

const getNextcloudParentFolder = (folder, rootFolderName) => {
  const parentFolder = getParentFolderFromPath(folder.path);
  if (parentFolder.path === '/') {
    return {
      _id: 'io.cozy.remote.nextcloud.files.root-dir',
      _type: 'io.cozy.remote.nextcloud.files',
      name: rootFolderName,
      path: '/',
      cozyMetadata: {
        sourceAccount: folder.cozyMetadata.sourceAccount
      }
    };
  } else {
    return {
      _type: 'io.cozy.remote.nextcloud.files',
      ...getParentFolderFromPath(folder.path),
      cozyMetadata: {
        sourceAccount: folder.cozyMetadata.sourceAccount
      }
    };
  }
};
```

then the usecallback is simplified to this
```js
const handleNavigateTo = useCallback(async () => {
  if (folder._id === ROOT_DIR_ID) {
    return navigateTo();
  }

  if (folder.dir_id) {
    const parentFolder = await getParentFolder(folder, client);
    return navigateTo(parentFolder);
  }

  if (
    folder._type === 'io.cozy.remote.nextcloud.files' &&
    (folder.path === '/' && showNextcloudFolder || showNextcloudFolder)
  ) {
    const nextcloudParentFolder = getNextcloudParentFolder(folder, rootFolderName);
    return navigateTo(nextcloudParentFolder);
  }
}, [client, folder, navigateTo, rootFolderName, showNextcloudFolder]);
```

Also it could be beneficial to add comments to better understand the business flow of what's going on

Reply:
I've tried to simplify the logic in stateless functions in the helper